api/controllers/case.js

```diff
@@ -191,7 +195,24 @@ router.put('/:caseId', function editCaseById (req, res) {
     }).then(function(data){
         let the_case = data[0];
         the_case.authors = data[1]; // authors
-        the_case.location = data[2]; // geolocation
+        let attachments = data[2]; // files and images
+        the_case.other_images = []
+        the_case.files = []
+        attachments.forEach(function(att){
+            if (att.type == 'file'){
+                the_case.files.push(att);
+            }else if (att.type == 'image'){
+                if (att.is_lead){
+                    the_case.lead_image = att;
+                }else{
+                    the_case.other_images.push(att);
+                }
+            }
+        });
+        the_case.methods = data[3]
```

Comment:
I agree, but in this case I don't have a good answer. I'd like to get more of the structure built directly in the SQL query if I can, but need to up my SQL game for that.